modules/telestion-services/src/main/java/de/wuespace/telestion/services/database/MongoDatabaseService.java

```diff
@@ -113,27 +128,120 @@ private void save(JsonMessage document) {
+	 * @param sort		@see {@link de.wuespace.telestion.services.database.MongoDatabaseService#setFindOptions(List, List)}.
+	 * @param limit		Limits the amount of returned entries. -1 equals all entries found.
+	 * @param skip		Specifies if and how many entries should be skipped.
+	 * @return			{@link io.vertx.ext.mongo.FindOptions} for the MongoClient.
+	 */
+	private FindOptions setFindOptions(List<String> fields, List<String> sort, int limit, int skip) {
+		return setFindOptions(fields, sort).setLimit(limit).setSkip(skip);
+	}
+
+	/**
+	 * Helper function to set the {@link io.vertx.ext.mongo.FindOptions}
+	 * for the {@link io.vertx.ext.mongo.MongoClient#findWithOptions(String, JsonObject, FindOptions)}.
+	 *
+	 * @param fields	List of key Strings in the collection limiting the fields that should be returned.
+	 * @param sort		List of key Strings that the returned data should be sorted by.
+	 * @return			{@link io.vertx.ext.mongo.FindOptions} for the MongoClient.
+	 */
+	private FindOptions setFindOptions(List<String> fields, List<String> sort) {
+		FindOptions findOptions = new FindOptions();
+		if (!fields.isEmpty()) {
+			JsonObject jsonFields = new JsonObject();
+			fields.forEach(f -> jsonFields.put(f, true));
+			findOptions.setFields(jsonFields);
+		}
+		if (!sort.isEmpty()) {
+			JsonObject jsonSort = new JsonObject();
+			sort.forEach(s -> jsonSort.put(s, -1));
+			findOptions.setSort(jsonSort);
+		}
+		return findOptions;
+	}
+
+	/**
+	 * Helper function to parse a query string to a {@link io.vertx.core.json.JsonObject}.
+	 *
+	 * @param query	JSON String - "{"key":"value"}, ..."
+	 * @return {@link io.vertx.core.json.JsonObject} query for
+	 * {@link io.vertx.ext.mongo.MongoClient#findWithOptions(String, JsonObject, FindOptions)}
+	 */
+	private JsonObject getJsonQueryFromString(String query) {
+		if (query.isEmpty()) {
+			return new JsonObject("{}");
+		} else {
+			try {
+				return new JsonObject(query);
+			} catch (DecodeException e) {
+				logger.error("No valid JSON String: ".concat(e.getMessage()).concat("\nReturning empty JsonObject."));
+				return new JsonObject();
+			}
+		}
+	}
+
+	/**
+	 * Helper function to convert a {@link java.util.Date} to a ISO-8601 Date/Time string.
+	 *
+	 * @param date	{@link java.util.Date} that should be converted.
+	 * @return	ISO-8601 Date/Time string representation
+	 */
+	private static String getISO8601StringForDate(Date date) {
+		DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.GERMANY);
+		dateFormat.setTimeZone(TimeZone.getTimeZone("CET"));
+		return dateFormat.format(date);
+	}
+
 	private static record Configuration(@JsonProperty JsonObject dbConfig, @JsonProperty String dbPoolName) {
 		private Configuration() {
-			this(new JsonObject().put("db_name", "raketenpraktikum").put("useObjectId", true), "raketenpraktikumPool");
+			this(new JsonObject().put("db_name", "daedalus2").put("useObjectId", true), "d2Pool");
```

Comment:
Any specific reason why you have daedalus2 in the private constructor? Not that it really matters, however for keeping this repository "client-free" you maybe could consider to unit the Configuration with `null` values

Reply:
Yes you're right. That still comes from my perception, that the private constructor is used, when no values are specified.